controllers/components/users/users.jsx

```diff
@@ -0,0 +1,217 @@
+            headers: {
+                'Content-Type': 'application/json',
+                'x-access-token': localStorage.token
+            },
+            method: 'POST',
+            body: JSON.stringify( data )
+        }).then( removePlayerSuccess )
+        .catch( that._errorHandler );
+
+        function removePlayerSuccess( res ) {
+            if ( !res.ok ) return that._errorHandler( res );
+
+            that.setState({
+                errorMessage: 'Player removed :(',
+                displayError: 'alert alert-success'
+            });
+        }
+    }
+
+    _pushPlayerToTeam( playerId, team ) {
+        const that = this;
+
+        let data = {
+            playerId: playerId,
+            teamId: team
+        };
+
+        fetch( '/api/teams/add/player', {
+            headers: {
+                'Content-Type': 'application/json',
+                'x-access-token': localStorage.token
+            },
+            method: 'POST',
+            body: JSON.stringify( data )
+        }).then( addPlayerSuccess )
+        .catch( that._errorHandler );
+
+        function addPlayerSuccess( res ) {
+            if ( !res.ok ) return that._errorHandler( res );
+
+            that.setState({
+                errorMessage: 'Player added :)',
+                displayError: 'alert alert-success'
+            });
+        }
+    }
+
+    _errorHandler( err, message ) {
+        const that = this;
+        if ( err.status >= 400 && err.status < 500 )
+            return err.text().then( function( message ) {
+                that.setState({
+                    errorMessage: message,
+                    displayError: 'alert alert-danger'
+                });
+            });
+        if ( message ) return that.setStatus({
+            errorMessage: message,
+            displayError: 'alert alert-danger'
+        });
+        alert( message );
+        return that.setState({
+            errorMessage: 'Sorry there\'s a problem with connection, please try again later',
+            displayError: 'alert alert-danger'
+        });
+    }
+
+    _addPlayerButton( user ) {
+        if ( localStorage.token ) {
+            const admin = JSON.parse( atob( localStorage.token.split( '.' )[ 1 ] ) );
+            if ( admin.isAdmin ) {
+                if ( admin.team ) {
+                    if ( !user.team ) {
+                        return <button className = 'btn btn-info btn-xs' onClick = { this._addPlayer } data-id = { user._id }>Add Player</button>;
+                    } else if ( user.team == admin.team ) {
+                        return <button className = 'btn btn-danger btn-xs' onClick = { this._removePlayer } data-id = { user._id }>Remove Player</button>;
+                    } else {
+                        return '';
+                    }
+                } else {
+                    return '';
+                }
+            } else {
+                return '';
+            }
+        } else {
+            return '';

```

Comment:
Really?! This looks ugly, fix it. ( identical code )

Reply:
Pyramid of doom 😄 
i will build the hole component from scratch.